### PR TITLE
Add support for NamedExpr in WrongInCompareTypeViolation

### LINKED ISSUE #1209: Ensure that `WrongInCompareTypeViolation` counts `NamedExpr`
Currently we don't handle `NamedExpr` in this violation.

One can write this code:

```python
if 'a' in (x := [1, 2, 'a', 3]):
    ...
```

And it won't be handled properly. 

File to fix: https://github.com/wemake-services/wemake-python-styleguide/blob/ea20c44bbbb8dcf70b3541b073cee25ae91a8354/wemake_python_styleguide/visitors/ast/compares.py#L454

We also need to test this violation. Make sure that you use `pytest.skipif(PY38, ...)` because this syntax is only valid starting from `python3.8+`

Related #1137

### PR DESCRIPTION
**Pull request by gillesigot**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #1209
- Refs #1137  

I tried something simpler with `_get_assigned_expr` function.  I'm still not sure whether I should keep it in the InCompareSanityVisitor class or not, to be used somewhere else...

### CODE REVIEW

File: wemake_python_styleguide/visitors/ast/compares.py [hunk 1]
```diff
@@ -455,5 +455,11 @@ def _check_single_item_container(self, node: ast.AST) -> None:
             self.add_violation(InCompareWithSingleItemContainerViolation(node))
 
     def _check_wrong_comparators(self, node: ast.AST) -> None:
+        node = self._get_assigned_expr(node)
         if isinstance(node, self._wrong_in_comparators):
             self.add_violation(WrongInCompareTypeViolation(node))
+
+    def _get_assigned_expr(self, node: ast.AST) -> ast.AST:
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/compares.py]: Let's better move this into `logic/walrus.py`

And one more thing: we use `compat/nodes.py` where we define nodes that appear in new `ast` releases.

File: tests/test_logic/test_walrus.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        '(x := 1)',
+        marks=pytest.mark.skipif(
+            sys.version_info < (3, 8),
+            reason='NamedExpr appeared in 3.8',
+        ),
+    ),
+])
+def test_get_assigned_expr(code):
+    """Returns the value of NamedExpr, the node itself otherwise."""
+    node = ast.parse(code)
```
Comment (sobolevn) [inline tests/test_logic/test_walrus.py]: We use `parse_ast_tree` fixture

File: wemake_python_styleguide/logic/walrus.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    If the node is an actual NamedExpr, the assigned value will be returned.
+    For other node type, the original node is returned.
+
+    This code is only executed on ``python3.8+``,
+    because before ``3.8.0`` release
+    there was no such thing as walrus operator.
+    """
+    if sys.version_info >= (3, 8):
+        if isinstance(node, ast.NamedExpr):  # pragma: no cover
```
Comment (sobolevn) [inline wemake_python_styleguide/logic/walrus.py]: We use `wemake_python_styleguide.compat.nodes.NamedExpr` for this case.
So, you don't have to use extra `if sys.version_info > ...`

Comment (sobolevn) [reply wemake_python_styleguide/logic/walrus.py]: By the way, we don't use `# pragma: no cover`
See https://sobolevn.me/2020/02/conditional-coverage

We use `# pragma: py-lt-38`

Comment (gillesigot) [reply wemake_python_styleguide/logic/walrus.py]: Ok, thx for quick review!

**Comment by sobolevn** [comment]
@gillesigot thanks!

You have to add `value: ast.expr` here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/compat/nodes.py#L21 Please, also make sure to ignore `WPS110` for this line. 

I removed this code for some reason several commits ago, sorry!

**Comment by sobolevn** [comment]
Thanks a lot, @gillesigot for your help!

And sorry for this long late-night debugging session in Github code editor you just had to witness.
But, I really need this logic merged, so other people can start working on other similar issues.

👍

**Comment by gillesigot** [comment]
> Thanks a lot, @gillesigot for your help!
> 
> And sorry for this long late-night debugging session in Github code editor you just had to witness.
> But, I really need this logic merged, so other people can start working on other similar issues.

@sobolevn Thank you! Sorry, I had a lot of things to finish on my side, so I was a bit late on this.
I would be glad to help on other PRs when I have time :slightly_smiling_face:

**Pull request merged**